plugins/woocommerce-admin/client/products/use-product-helper.ts

```diff
@@ -152,44 +183,57 @@ export function useProductHelper() {
 			return updateProduct( productId, {
 				...product,
 				status,
-			} ).then(
-				( updatedProduct ) => {
-					if ( ! skipNotice ) {
-						const noticeContent =
-							product.status === 'draft' &&
-							updatedProduct.status === 'publish'
-								? __( 'Product published.', 'woocommerce' )
-								: __(
-										'Product successfully updated.',
-										'woocommerce'
-								  );
-						createNotice( 'success', `🎉‎ ${ noticeContent }`, {
-							actions: getNoticePreviewActions(
-								updatedProduct.status,
-								updatedProduct.permalink
-							),
+			} )
+				.then( async ( updatedProduct ) =>
+					updateVariationOrders(
+						updatedProduct.id,
+						( product as ProductVariationOrders ).variationOrders
+					)
+						.then( () =>
+							invalidateResolutionForStoreSelector(
```

Comment:
If we can introduce the `batchUpdateProductVariationsSuccess` action above, we might not need to invalidate this selector anymore.